SI/Sem.04/Pract.04/Task12.cpp

```diff
@@ -0,0 +1,38 @@
+#include <iostream>
+
+bool isPowerOfTwo(int n) {
+	while (n != 0) {
+		if (n == 1) {
+			return true;
+		}
+		if (n % 2 == 1) {
+			return false;
+		}
+		n /= 2;
+	}
+	return false;
+}
+int nearestPowerOfTwo(int n) {
+	int left = n;
+	while (!isPowerOfTwo(left)) {
```

Comment:
Can be optimized. This way you will check every number if it is power of two.
Example you enter 333. You will check all the numbers from 333 to 256 if they are power of two and then all the numbers from 333 to 512. Which is not optimal